metals/src/main/scala/scala/meta/internal/metals/doctor/ProblemResolver.scala

```diff
@@ -188,11 +178,12 @@ class ProblemResolver(
       mtagsResolver.isSupportedScalaVersion(version)
 
     val scalaVersionProblem = scalaTarget.scalaVersion match {
-      case version if !isSupportedScalaVersion(version) && scalaTarget.isSbt =>
-        if (ScalaVersions.isFutureVersion(version))
-          Some(FutureSbtVersion)
-        else
-          Some(UnsupportedSbtVersion)
+      case _ if scalaTarget.isSbt =>
+        scalaTarget.sbtVersion match {
+          case Some(version) if !SemVer.isCompatibleVersion("1.3.2", version) =>
+            Some(UnsupportedSbtVersion(version))
+          case _ => None
+        }
```

Comment:
Looks like this case makes the following cases (like `case version if !isSupportedScalaVersion(version) =>` unreachable for sbt project.

I guess we need to move the `scalaTarget.sbtVersion` check into `case` like 

```scala
case _ if scalaTarget.sbtVersion match { ... } => 
```

Reply:
Or maybe `findProblem` doesn't need to return `Option[ScalaProblem]`, but it can return something like `List[ScalaProblem]` 🤔 , because sbt  deprecation problem and unsupported scalaversion problems are independent problems.
What do you think about it? @tgodzik @kpodsiad (someone who might be familiar with Metals Doctor)

Anyway, you don't need to work on it in this PR 👍

Reply:
Ok so if I understand correctly, when a target has `isSbt` the `scalaVersion` is not the project's Scala version but actually the Scala version that sbt was built on. In my case, my project uses Scala 2.12.14 (not deprecated by Metals) but sbt 1.3.12 uses Scala 2.12.10 (deprecated by Metals). So when checking the sbt Scala version, we want to surface a message to the user that says "Please upgrade sbt" even though the underlying problem is sbt's Scala version.

> maybe findProblem doesn't need to return Option[ScalaProblem], but it can return something like List[ScalaProblem]  

I think it is an `Option` because each target can have one recommendation. So you could have a Scala target that says "Please upgrade Scala" and the sbt target that says "Please upgrade sbt" but each target can only say one thing.